### PR TITLE
Use prettier for autoformatting

### PR DESCRIPTION
**Pull request by Graham42**
Because auto-formatting is a liberating experience. Free to just code,
not think about style.

For more reasons see: https://github.com/prettier/prettier#why-prettier

### CODE REVIEW

**Comment by dthree** [comment]
Seems to match my preferred code style pretty closely, so I wouldn't mind trying it out.

What is the general cycle -- run an NPM script before publish and it edits the source directory?

**Comment by Graham42** [comment]
In the dev environment, on npm install, `husky` will set up a precommit hook to run lint-staged. This is set up by the `precommit` script in package.json.
Then, `lint-staged` will format any `.js` files that you've staged for your git commit.

Workflow is just as normal:
1. Change `js` file
2. `git add`
3. `git commit`

Alternatively you can install one of the editor integrations to have the code autoformat on save https://prettier.io

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+lib-cov
+
+# Coverage directory used by tools like istanbul
+coverage
+
+# nyc test coverage
+.nyc_output
+
+# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
+.grunt
```
Comment (dthree) [inline .gitignore]: Is there a reason you added all these items?

**Comment by dthree** [comment]
Looking good! One last q.

File: .gitignore (hunk 1 above)
Comment (Graham42) [inline .gitignore]: I just copied the nodejs gitignore template from https://github.com/github/gitignore/blob/master/Node.gitignore

**Comment by Graham42** [comment]
Rebased to resolve conflicts. Was all just the formatting changes.

**Pull request merged**